addons/vitals/functions/fnc_handleUnitVitals.sqf

```diff
@@ -147,29 +147,29 @@ _unit setVariable [VAR_BLOOD_PRESS, _bloodPressure, _syncValues];
 _bloodPressure params ["_bloodPressureL", "_bloodPressureH"];
 
 // Statements are ordered by most lethal first.
-// Add SpO2 reactions to switch statement ---------------------------------------------------------------------
- switch (true) do {
-    case (_spo2 < EGVAR(breathing,SpO2_dieValue) && EGVAR(breathing,SpO2_dieActive)): {
+switch (true) do {
+    case ((_spo2 < EGVAR(breathing,SpO2_dieValue)) && EGVAR(breathing,SpO2_dieActive)): {
         TRACE_3("O2 Fatal",_unit,EGVAR(breathing,SpO2_dieValue),_spo2);
         [QACEGVAR(medical,FatalInjury), _unit] call CBA_fnc_localEvent;
+        [QACEGVAR(medical,Death), _unit] call CBA_fnc_localEvent; // We have to call the Death state because if FatalInjuries is disabled, the patient won't actually die and just go into arrest
```

Comment:
this is intended behaviour

Reply:
Not if SpO2 die value is enabled. If players are not set to die due to fatal injuries, then once they pass the fatal O2 limit, they will just go into arrest but never actually die

Reply:
Yes, because fatal injuries when selected as none dont kill unit

Reply:
also this event does not even exist in statemachine, all it does is resetting volume/hearing